lib/tasks/version_bump.rake

```diff
@@ -0,0 +1,425 @@
+  commits.each(&:perform!)
+
+  git("push", "-f", "--set-upstream", "origin", branch)
+
+  make_pr(
+    base: base,
+    branch: branch,
+    title:
+      "Version bump#{"s" if commits.length > 1} for #{base}: #{commits.map { |c| "v#{c.version}" }.join(", ")}",
+  )
+end
+
+def make_pr(base:, branch:, title:)
+  params = { expand: 1, title: title, body: <<~MD }
+      > :warning: This PR should not be merged via the GitHub web interface
+      > 
+      > It should only be merged (via fast-forward) using the associated `bin/rake version_bump:*` task.
+    MD
+
+  if !test_mode?
+    system(
+      "open",
+      "https://github.com/discourse/discourse/compare/#{base}...#{branch}?#{params.to_query}",
+      exception: true,
+    )
+  end
+
+  puts "Do not merge the PR via the GitHub web interface. Get it approved, then come back here to continue."
+end
+
+def fastforward(base:, branch:)
+  if dry_run?
+    puts "[DRY RUN] Skipping fastforward of #{base}"
+    return
+  end
+
+  confirm "Ready to merge? This will fast-forward #{base} to #{branch}"
+  begin
+    git "push", "origin", "#{branch}:#{base}"
+  rescue => e
+    raise <<~MSG
+      #{e.message}
+      Error occured during fastforward. Maybe another commit was added to `#{base}` since the PR was created, or maybe the PR wasn't approved.
+      Don't worry, this is not unusual. To update the branch and try again, rerun this script again. The existing PR and approval will be reused.
+    MSG
+  end
+  puts "Merge successful"
+end
+
+def stage_tags(commits)
+  puts "Staging tags locally..."
+  commits.each do |commit|
+    commit.tags.each { |tag| git "tag", "-f", "-a", tag.name, "-m", tag.message, commit.ref }
+  end
+end
+
+def push_tags(commits)
+  tag_names = commits.flat_map { |commit| commit.tags.map(&:name) }
+
+  if dry_run?
+    puts "[DRY RUN] Skipping pushing tags to origin (#{tag_names.join(", ")})"
+    return
+  end
+
+  confirm "Ready to push tags #{tag_names.join(", ")} to origin?"
+  tag_names.each { |tag_name| git "push", "-f", "origin", "refs/tags/#{tag_name}" }
+end
+
+def with_clean_worktree(origin_branch)
+  origin_url = git("remote", "get-url", "origin").strip
+
+  if !test_mode? && !origin_url.include?("discourse/discourse")
+    raise "Expected 'origin' remote to point to discourse/discourse (got #{origin_url})"
+  end
+
+  git "fetch", "origin", origin_branch
+  path = "#{Rails.root}/tmp/version-bump-worktree-#{SecureRandom.hex}"
+  begin
+    FileUtils.mkdir_p(path)
+    git "worktree", "add", path, "origin/#{origin_branch}"
+    Dir.chdir(path) { yield } # rubocop:disable Discourse/NoChdir
+  ensure
+    puts "Cleaning up temporary worktree..."
+    git "worktree", "remove", "--force", path, silent: true, allow_failure: true
+    FileUtils.rm_rf(path)
+  end
+end
+
+desc "Stage commits for a beta version bump (e.g. beta1.dev -> beta1 -> beta2.dev). A PR will be created for approval, then the script will prompt to perform the release"
+task "version_bump:beta" do
```

Comment:
When reviewing, it probably makes sense to read through each of the 'task' definitions first, and then look at the helper methods if/when needed